api/src/main/java/org/apache/iceberg/transforms/Transforms.java

```diff
@@ -78,9 +107,11 @@ private Transforms() {}
    * @param type the {@link Type source type} for the transform
    * @param <T> Java type passed to this transform
    * @return an identity transform
+   * @deprecated use {@link #identity()} instead; will be removed in 2.0.0
    */
+  @Deprecated
```

Comment:
To ensure that everything works, I removed the deprecated methods in this class and made sure that all tests in core were passing. Then I added these back marked deprecated (so that Spark 3.2 and other versions would still work).